tensorflow/python/ops/numpy_ops/np_array_ops.py

```diff
@@ -819,6 +819,11 @@ def f(x):
         x = array_ops.where_v2(x < 0, np_utils.add(x, a_rank), x)
       return x
     return nest.map_structure(f, axes)
+  
+  if axis2 < -array_ops.rank(a) or axis2 >= array_ops.rank(a):
```

Comment:
This fails if these are tensors, since you can't use a tensor as a python bool.   You probably need to use something like this: https://github.com/tensorflow/tensorflow/blob/517df2d259ba27f2ef3da399bf9c3b081d7a57b7/tensorflow/python/ops/image_ops_impl.py#L71

though we shouldn't add the dependency on image_ops_impl, so maybe re-implement it, or find another similar function in a shared utility